mlrun/artifacts/base.py

```diff
@@ -1032,3 +1033,43 @@ def generate_target_path(item: Artifact, artifact_path, producer):
             suffix = f".{item.format}"
 
     return f"{artifact_path}{item.key}{suffix}"
+
+
+def convert_legacy_artifact_to_new_format(
+    legacy_artifact: typing.Union[LegacyArtifact, dict]
+) -> Artifact:
+    """Converts a legacy artifact to a new format.
+
+    :param legacy_artifact: The legacy artifact to convert.
+    :return: The converted artifact.
+    """
+    if isinstance(legacy_artifact, LegacyArtifact):
+        legacy_artifact_dict = legacy_artifact.to_dict()
+    elif isinstance(legacy_artifact, dict):
+        legacy_artifact_dict = legacy_artifact
+    else:
+        raise TypeError(
+            f"Unsupported type '{type(legacy_artifact)}' for legacy artifact"
+        )
+    artifact_dict = {
+        "metadata": {},
+        "spec": {},
+        "status": {},
+        "kind": legacy_artifact_dict.get("kind", "artifact"),
+    }
+    for key, value in legacy_artifact_dict.items():
+        if value is not None:
+            if key in mlrun.artifacts.base.ArtifactMetadata._dict_fields:
+                artifact_dict["metadata"][key] = value
+            elif (
+                key
+                in mlrun.artifacts.base.ArtifactSpec._dict_fields
+                + mlrun.artifacts.base.ArtifactSpec._extra_fields
```

Comment:
Well... Almost 😢. Note that there are other classes inheriting from `ArtifactSpec` which potentially add more fields, for example take a look at `DatasetArtifactSpec`, `ModelArtifactSpec` etc. 
You need to use the `kind` of the artifact to retrieve the correct artifact type (there's a mapping called `artifact_types` in the artifact-manager which maps kind to artifact type), and take the spec and its `_dict_fields` from there...
BTW, one other way to do this would be to just do a `from_dict` with the spec class using the dictionary you have in your hand. Same with `ArtifactMetadata.from_dict(legacy_artifact_dict)` etc. Then construct your artifact with those results (and add root-level fields, which should be just `kind`).